libbeat/cmd/instance/beat_test.go

```diff
@@ -92,3 +95,37 @@ func TestInitKibanaConfig(t *testing.T) {
 	assert.Equal(t, "https", protocol)
 	assert.Equal(t, "127.0.0.1:5601", host)
 }
+
+func TestEmptyMetaJson(t *testing.T) {
+	b, err := NewBeat("filebeat", "testidx", "0.9")
+	if err != nil {
+		panic(err)
+	}
+
+	metaPath := paths.Resolve(paths.Data, "meta.json")
+	backupMetaPath := paths.Resolve(paths.Data, "meta.json.backup")
+
+	// backup existing meta file
+	_, err = os.Stat(metaPath)
+	backedUp := false
+	if !os.IsNotExist(err) {
+		err := os.Rename(metaPath, backupMetaPath)
+		assert.Equal(t, nil, err, "Cannot backup meta file")
+		backedUp = true
+	}
+
+	// prepare empty meta
+	ioutil.WriteFile(metaPath, nil, 0644)
+
+	metaErr := b.loadMeta()
+
+	// cleanup
+	os.Remove(metaPath)
+	if backedUp {
+		err = os.Rename(backupMetaPath, metaPath)
+		assert.Equal(t, nil, err, "Cleanup failed")
+	}
+
+	assert.Equal(t, nil, metaErr, "Unable to load meta file properly")
+	assert.NotEqual(t, uuid.Nil, b.Info.ID, "Beats UUID is not set")
+}
```

Comment:
Currently we do not execute tests in parallels, every time that we rely on the filesystem for testing this could lead to races condition when we moved to that.

I see two things we could do:
- Either we write a new system test to validates the issue, they are run in isolation and slower.
- Or we make the `loadMeta(metapath string)` take an arguments instead and test with a random file.

Also, looking at the original issue did we find out why this is happening?
Also you will need to add an item to the CHANGELOG.next.md as a bugfix.

Reply:
good point, i worked on an assumption that this will not be run in parallel, better safe than sorry
will go with filename arg to`loadMeta` func to keep tests short.
regarding root cause nothing found yet